network/transport/grpc/connection_test.go

```diff
@@ -139,8 +123,8 @@ func Test_conn_startSending(t *testing.T) {
 			return atomic.LoadInt32(&connection.activeGoroutines) == 0, nil
 		}, 5*time.Second, "waiting for all goroutines to exit")
 
-		// err status is set on connection. Due to EOF it's an unknown error
-		assert.Equal(t, codes.Unknown, connection.status.Load().Code())
+		// Last received message is dropped and no status is set. Default value is OK.
```

Comment:
This is a bit weird, status is now the same as being operational. Shouldn't it be a disconnected status somehow?

Reply:
I don't know what the right solution is here.

connection.status is a grpc Status that reflects the state of the last message received. (it is a bit of a hack for the backoff) no-status is considered equal to `OK` (this situation I think)

When the connection's context is cancelled it means that the receiving side closed the connection. `conn.startReceiving` ignores any future message (including any possible errors) in this case and the status of the last processed incoming message was still ok. 

Also, `Unauthenticated` is the only status we act on and does not apply here, all other statuses are handled the same as the current situation.

Reply:
This test is weird anyway. It supposedly tests `conn.startSending`, but the only thing it tests is error handling of `conn.startReceiving` after disconnect with a stub stream that simulates stream closure with an EOF message